tfexec/internal/e2etest/util_test.go

```diff
@@ -63,12 +89,12 @@ func runTestVersions(t *testing.T, versions []string, fixtureName string, cb fun
 				os.RemoveAll(td)
 			})
 
-			// TODO: do this in a cleaner way than string comparison?
 			var execPath string
-			switch {
-			case strings.HasPrefix(tfv, "refs/"):
+			if localBinPath := os.Getenv("TFEXEC_E2ETEST_TERRAFORM_PATH"); localBinPath != "" {
+				execPath = localBinPath
+			} else if strings.HasPrefix(tfv, "refs/") {
```

Comment:
I'm wondering if long-term we could just get rid of this entirely and assume that if Terraform is built from scratch, then it's done outside of tfexec and path to the binary is passed via `TFEXEC_E2ETEST_TERRAFORM_PATH`? 🤔 

Not sure how many tests actually still use this though.

Reply:
Yeah - I admit it would be a lot to ask as part of this PR, but it just seemed appealing from the perspective of reducing the surface of the library, if building it outside is an option now with your new ENV variable.

Reply:
Opened https://github.com/hashicorp/terraform-exec/issues/318